Add Travis check for YAML_TEST_DATA in yaml schedule

sofiasyria opened the pull request:
Amplification of Travis checks in order to cover scenario where YAML_TEST_DATA variable is defined in yaml schedule's vars, to point to test data file.

- Related ticket: https://progress.opensuse.org/issues/64752
- Verification run: 
   - [Remove/rename references modules or test_data](https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/jobs/670154270?utm_medium=notification&utm_source=github_status)
   - [Referenced non-existent test data](https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/jobs/669651402?utm_medium=notification&utm_source=github_status)

rwx788 commented on Makefile:
```diff
@@ -120,17 +120,13 @@ perlcritic: tools/lib/
 .PHONY: detect-nonexistent-testdata
 detect-nonexistent-testdata:
-	tools/detect_nonexistent_testdata `git diff --exit-code $$(git merge-base master HEAD) | grep '+  $include: test_data/' | grep 'yaml$$' | awk '{print $$3}'`
+	export PERL5LIB=${PERL5LIB_} ; tools/detect_nonexistent_testdata `git diff --name-only --exit-code $$(git merge-base master HEAD) | grep '^schedule/*'`
```
As we have fixed similar issue recently, let's filter out deleted schedules. We also could have checked yaml/yml files only in schedule directory, even though there should be nothing eles

sofiasyria replied:
Good point for the deleted schedules. The second sentence confuses me a bit. You mean, I should add "| grep "ml$" at the end of line 129 ?

rwx788 replied:
We could have `'^schedule/.*\.y.?ml$'` in single grep instead of 2 of them.

sofiasyria replied:
done

rwx788 commented on tools/test_deleted_renamed_referenced_files:
```diff
@@ -0,0 +1,27 @@
+    filename=$(echo $FILE | cut -f 2- -d '/' | cut -f 1 -d '.')
+    target_paths='schedule/ products/*/main.pm lib/main_common.pm'
+    # In case file is test_data yaml file, the filename given is the same as in schedule files
+    elif [ -n "$(echo $FILE | grep '.yaml$')" ]; then
```
Some teams use `yml` exstentions, so I would propose using `''\.y.\?ml$'`.

sofiasyria replied:
Done

rwx788 commented on tools/test_deleted_renamed_referenced_files:
```diff
@@ -0,0 +1,27 @@
+    fi
+    if [ -n "$(echo $FILE | grep '.pm$')" ]; then
+    # In case file is a module, module name appears in scheduling files excluding 'tests/' and file extension
+    filename=$(echo $FILE | cut -f 2- -d '/' | cut -f 1 -d '.')
```
I guess we could use basename/dirname utils here

sofiasyria replied:
hmm... but we don't use basename or dirname in schedule. We just cut the tests/ and the .pm 
tests/installation/partitioning.pm -> installation/partitioning

rwx788 replied:
Ah, it's variable name which confused me, I thought that you are extracting just file name.
Also, consider this example: 
```sh
echo "inst/pam.pam.pm" | cut -f 1 -d '.'
Output: inst/pam
```
As we know that we won't have extensions like `tar.gz` we could simply remove `.pm` from the end.

With sed:
```
echo tests/installation/part.pm.pm | sed -E "s/^tests\/(.*)\.pm$/\1/g"
```

So in this case I would suggest at least renaming `filename` variable to `module`.
Could you also please fix indentation in the if clause?
Also, keep in mind that normally you put `then` on next line and semicolon syntax is more for one liners ( see examples here: https://www.tutorialspoint.com/unix/if-else-statement.htm )

sofiasyria replied:
> So in this case I would suggest at least renaming `filename` variable to `module`.

Initially it was 'module' in Joaquin's script, I changed it to 'filename' because now the variable might represent either a module or a test_data file. After your comment I changed it to file_to _verify in order to avoid confusion. WDYT?

Rest, done.

rwx788 replied:
Good point. I guess we can leave current name then. Cheers!

rwx788 commented on tools/test_deleted_renamed_referenced_files:
```diff
@@ -0,0 +1,30 @@
+    if [ -n "$(echo $FILE | grep '\.pm$')" ] 
+    then
+	# In case file is a module, module name appears in scheduling files excluding 'tests/' and file extension
+	file_to_verify=$(echo $FILE | sed -E 's/^tests\/(.*)\.pm$/\1/g')
```
Let's fix indentation on these 2 lines too.

sofiasyria replied:
For some weird reason, these two line appear to be a bit to the left compared to rest, but when I press "view file" they look aligned with the rest. In my local repo they also look aligned.

rwx788 replied:
Looks like it's tab vs spaces issue. On the lines above you have spaces and these 2 have tabs. Please, use one of two and use it consistently in the script. We also can have 4 hours discussion about tabs vs spaces and if tab should be 2 or 4 spaces =D

sofiasyria replied:
Ahh.. no, there are much better ways to spend 4 hours. So Indeed, it looks like when using tab in particular page "Files changes" of the pull request, indentation looks different. So I used space everywhere.

rwx788 commented:
Just to be sure, have you tested latest version? As travis runs are from yesterday and last push was 3 hours ago. Static code analysis is good, but running still can show things we have missed =).

sofiasyria commented:
> Just to be sure, have you tested latest version? As travis runs are from yesterday and last push was 3 hours ago. Static code analysis is good, but running still can show things we have missed =).

Today I changed the tabs to spaces in tools/test_deleted_renamed_referenced_files . I just rerun the validation test that concerns that part of the pull request and replaced the verification link on the description.

rwx788 commented:
Breaks in case no test data is in the schedule: https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/jobs/670463776?utm_medium=notification&utm_source=github_status 
Reverting to fix without stress.
